client/core/trade.go

```diff
@@ -893,7 +893,18 @@ func (t *trackedTrade) isSwappable(ctx context.Context, match *matchTracker) boo
 				match.MetaData.Proof.SelfRevoked = true
 				return false
 			}
-			return ready
+			if !ready {
+				return false
+			}
+			// Re-audit the contract now that we know it is mined
+			// and has the required confirmations.
+			if err := t.reAuditContract(match); err != nil {
+				t.dc.log.Errorf("Match %s not swappable: repeat audit of maker's contract failed: %v",
+					match, err)
+				match.swapErr = fmt.Errorf("Counter-party contract repeat audit error: %v", err)
+				return false
+			}
```

Comment:
I really don't think we need to do this. I understand the compulsion, but unless we're willing to say that the attacker can spoof a transaction hash, this doesn't really do anything for us. And the way I understand it, spoofing a transaction hash is realistically impossible. 

The fact that we're using `AuditContract` to also find the transaction on-chain causes a lot of ambiguity around the roles these methods perform, and results in extra work. We could have a cleaner, more user-friendly interface if we just define `AuditContract` to audit the server-supplied tx data, and `SwapConfirmation` to deal with the blockchain.

Reply:
Works for me. @chappjc?

Reply:
Yes, agree.